tests/specs/lemmas.k

```diff
@@ -65,6 +72,29 @@ module LEMMAS [symbolic]
     rule WS         ++ .ByteArray => WS  [simplification]
     rule .ByteArray ++ WS         => WS  [simplification]
 
+    // #range
+
+    rule #range(_, _, K) => 
+         .ByteArray 
+      requires notBool K >Int 0 [simplification]
+
+    rule #range(M [ N := BUF:ByteArray ], L, K) => 
+         #range(M, L, minInt(K, N -Int L)) ++ #range(M [ N := BUF ], N, K -Int minInt(K, N -Int L)) 
+      requires K  >Int 0 andBool L <Int N [simplification]
+
+    rule #range(M [ N := BUF:ByteArray ], L, K) => 
+           BUF [ L -Int N .. minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) ]
+           ++ #range(M, N +Int #sizeByteArray(BUF), K -Int minInt(K, #sizeByteArray(BUF) -Int (L -Int N)))
+      requires K  >Int 0 andBool L <=Int N +Int #sizeByteArray(BUF) andBool L >=Int N [simplification]
```

Comment:
Probably also reads nicer with `#let` binding. They can be nested too.

Reply:
https://github.com/runtimeverification/k/blob/382a862f0fe5fd1ca00c4c5d187939db2b4b0506/k-distribution/tests/regression-new/let-test/test.k#L10

Reply:
maybe something like:

```
#let S = L -Int N #in
#let W = minInt(K, #sizeByteArray(BUF) -Int (L -Int N)) #in
  ...
``

Reply:
I changed this to use nested let bindings as suggested.